### PR TITLE
Remove sending extra funds to sender

### PR DESCRIPTION
**Pull request by karankurbur**
Remove sending back stuck funds in batcher

Currently, sending extra funds to batcher will send all funds that are
in the batcher contract to msg.sender. We replace this action by
letting an admin call the recover function to remove stuck funds.
This is done to prevent someone from front running a transaction
that leaves ETH in the contract and calling batch() right after to
withdraw the contract balance.

CLOSES Ticket: BG-28164

### CODE REVIEW

File: test/batcher.js [hunk 1]
```diff
@@ -197,338 +197,410 @@ contract("Batcher", (accounts) => {
...
+  //       values: [5],
+  //       expectedRetVal: 5,
+  //       expectOverallFailure: true,
+  //       expectedErrMsg: sendFailedErrorMsg
+  //     };
+  //     await runTestBatcherDriver(params);
+  //   });
+  // });
+
+  // describe('Good transactions multiple recipients', () => {
```
Comment (wesgraham) [inline test/batcher.js:263]: What are we doing with all these comments here?

File: contracts/Batcher.sol [hunk 2]
```diff
@@ -49,10 +48,11 @@ contract Batcher {
     /**
      * Transfer funds in a batch to each of recipients
      * @param recipients The list of recipients to send to
-     * @param values The list of values to send to recipients. 
+     * @param values The list of values to send to recipients.
      *  The recipient with index i in recipients array will be sent values[i].
      *  Thus, recipients and values must be the same length
      */
+
```
Comment (marktoda) [inline contracts/Batcher.sol:55]: remove extra \n

File: test/batcher.js [hunk 3]
```diff
@@ -65,18 +64,19 @@ const assertBalanceDiff = (start, end, diff, errMsg) => {
...
   let gasHeavyInstance;
+  let testBatcherDriverInstance;
 
   const sender = accounts[0];
   const batcherOwner = accounts[8];
-  const zeroAddr = "0x0000000000000000000000000000000000000000";
+  const zeroAddr = '0x0000000000000000000000000000000000000000';
 
-  before(async () => {
+  beforeEach(async () => {
```
Comment (marktoda) [inline test/batcher.js:79]: Why beforeEach? Should only need to instantiate these once

File: test/batcher.js (hunk 1 above)
Comment (marktoda) [reply test/batcher.js:263]: ^ yeah, please don't remove coverage.

File: test/batcher.js [hunk 4]
```diff
@@ -197,338 +197,410 @@ contract("Batcher", (accounts) => {
...
+  //       // and then another 10,000 gas for each subsequent iteration
+  //       gasLimit: 8e4
+  //     };
+  //     await runTestBatcherDriver(params);
+  //   });
+  // });
+  //
+  describe('Only owner functions', () => {
+    const recoverAddress = '0xeE223B2C6E49AE6F235A2B928c355454152d6ecc';
+
```
Comment (marktoda) [inline test/batcher.js:533]: Please add coverage for recovering USDT

**Review by marktoda — CHANGES_REQUESTED** [review]
Thanks for running linter. Looks largely good. A couple comments

File: test/batcher.js (hunk 1 above)
Comment (karankurbur) [reply test/batcher.js:263]: sorry, commented out to run faster locally haha

File: contracts/Batcher.sol (hunk 2 above)
Comment (marktoda) [reply contracts/Batcher.sol:55]: ?

**Review by wesgraham — APPROVED** [review]
Lgtm, nice work

File: test/batcher.js [hunk 5]
```diff
@@ -694,7 +771,42 @@ contract("Batcher", (accounts) => {
...
+        const batcherBalance = await tetherTokenContract.balanceOf.call(batcherInstance.address);
+        const senderBalance = await tetherTokenContract.balanceOf.call(batcherOwner);
+
+        assert.strictEqual(
+          batcherBalance.toString(),
+          '0',
+          'Batcher USDT balance should be 0.'
+        );
+        // assert.strictEqual(
```
Comment (marktoda) [inline test/batcher.js]: whats up with this commented out assert?

File: test/batcher.js [hunk 6]
```diff
@@ -694,7 +771,42 @@ contract("Batcher", (accounts) => {
...
+          from: batcherOwner
+        });
+
+        await tetherTokenContract.transfer(batcherInstance.address, 100, {
+          from: batcherOwner
+        });
+
+        const tokenTransferData = getUSDTTokenTransferData(batcherOwner, 100);
+
+        // console.log(tokenTransferData);
```
Comment (marktoda) [inline test/batcher.js]: remove

**Comment by marktoda** [comment]
Please also improve commit message to explain the change you made further and why we are making it.

File: test/batcher.js [hunk 7]
```diff
@@ -197,88 +205,88 @@ contract("Batcher", (accounts) => {
...
-      testBatcherBalance,
+      parseInt(testBatcherBalance),
       expectedRetVal,
-      "TestBatcherDriver has more or less funds left over than it should"
+      'Sender should not receive funds unless it is a recipient.'
     );
-  };
 
-  afterEach(async () => {
+    //Check batcher contract balances
```
Comment (marktoda) [inline test/batcher.js:230]: nit space after comment

File: test/batcher.js [hunk 8]
```diff
@@ -197,88 +205,88 @@ contract("Batcher", (accounts) => {
...
       };
       await runTestBatcherDriver(params);
     });
 
-    it("Correctly sends with extra value", async () => {
+    it('Correctly sends with extra value', async () => {
       const params = {
         recipients: accounts.slice(1, 4),
-        values: createRandIntArr(3),
+        values: [5, 5, 10],
```
Comment (marktoda) [inline test/batcher.js:283]: Why are you removing the randomness here? Randomness allows for some edge cases to be [potentially] captured automatically

**Review by marktoda — COMMENTED** [review]
Improve commit message please. Short summary then long description. Not readable in `git log` as-is. See https://chris.beams.io/posts/git-commit/

**Review by marktoda — APPROVED** [review]
LGTM
